pkg/liquidity-mining/contracts/test/MockGaugeController.sol

```diff
@@ -65,11 +63,33 @@ contract MockGaugeController is IGaugeController {
         // solhint-disable-previous-line no-empty-blocks
     }
 
-    function gauge_relative_weight(address, uint256) external view override returns (uint256) {
-        // solhint-disable-previous-line no-empty-blocks
+    function gauge_relative_weight(address gauge, uint256) external view override returns (uint256) {
+        return _weights[gauge];
     }
 
     function change_type_weight(int128, uint256) external override {
         // solhint-disable-previous-line no-empty-blocks
     }
+
+    function add_gauge_with_weight(
+        address gauge,
+        int128 gaugeType,
+        uint256 weight
+    ) external {
+        _add_gauge(gauge, gaugeType, weight);
+    }
+
+    function _add_gauge(
+        address gauge,
+        int128 gaugeType,
+        uint256 weight
+    ) internal {
+        require(!_validGauge[gauge], "Gauge already exists on controller");
+        require(gaugeType >= 0 && gaugeType < _numGaugeTypes, "Invalid gauge type");
+        _validGauge[gauge] = true;
+        if (weight > 0) {
+            _weights[gauge] = weight;
+        }
+        emit NewGauge(gauge, gaugeType, weight);
+    }
```

Comment:
I'd keep the old `add_gauge`, and instead of having `_with_weight` have `setGaugeWeight` instead.

Reply:
Done.